geonode/geoserver/manager.py

```diff
@@ -437,108 +442,109 @@ def set_permissions(
         approval_status_changed: bool = False,
         group_status_changed: bool = False,
     ) -> bool:
+
         _resource = instance or ResourceManager._get_instance(uuid)
 
         try:
             if _resource:
                 _resource = _resource.get_real_instance()
-                logger.error(
-                    f"Fixup GIS Backend Security Rules Accordingly on resource {instance} {isinstance(_resource, Dataset)}"
-                )
+                logger.info(f'Requesting GeoFence rules on resource "{_resource}" :: {type(_resource).__name__}')
                 if isinstance(_resource, Dataset):
-                    if settings.OGC_SERVER["default"].get("GEOFENCE_SECURITY_ENABLED", False):
+                    if settings.OGC_SERVER["default"].get("GEOFENCE_SECURITY_ENABLED", False) or getattr(
+                        settings, "GEOFENCE_SECURITY_ENABLED", False
+                    ):
                         if not getattr(settings, "DELAYED_SECURITY_SIGNALS", False):
-                            _disable_cache = []
+                            batch = AutoPriorityBatch(
+                                gf_utils.get_first_available_priority(), f"Set permission for resource {_resource}"
+                            )
+
+                            workspace = get_dataset_workspace(_resource)
+
+                            if not created:
+                                gf_utils.collect_delete_layer_rules(workspace, _resource.name, batch)
+
+                            exist_geolimits = None
```

Comment:
i would suggest to have it as "False" and not None since the name say "exists"

Reply:
We need the 3 different values:

- true -> limit exists
- false -> limit does not exist
- None -> not computed